src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -1531,8 +1531,28 @@ void MacroAssembler::check_klass_subtype_slow_path(Register sub_klass,
 
   br(Assembler::NE, *L_failure);
 
-  // Success.  Cache the super we found and proceed in triumph.
-  str(super_klass, super_cache_addr);
+  // Success. Try to cache the super we found and proceed in triumph.
+
+  uint32_t super_cache_backoff = checked_cast<uint32_t>(SecondarySuperMissBackoff);
+  if (super_cache_backoff > 0) {
+    Label L_skip;
+
+    ldrw(rscratch1, Address(rthread, JavaThread::backoff_secondary_super_miss_offset()));
+    subsw(rscratch1, rscratch1, 1);
+    br(Assembler::GT, L_skip);
```

Comment:
```suggestion
    subw(rscratch1, rscratch1, 1);
    cbzw(rscratch1, 31, L_skip);
```

I know this is >= 0 rather than > 0, but that doesn't matter, and we should make this code small.

Reply:
Did you mean `tbz(rscratch1, 31, L_skip);` here?

Reply:
Did `tbz` in the new commit. AFAICS, this indeed makes the flag restoration unnecessary, removed those too. Testing now...

Reply:
> Did you mean `tbz(rscratch1, 31, L_skip);` here?

Argh, of course. Yes.